src/util/fields.py

```diff
@@ -1122,13 +1122,62 @@ def change_field_selection_values(cr, model, field, mapping, skip_inherit=()):
     table = table_of_model(cr, model)
 
     if column_exists(cr, table, field):
-        query = "UPDATE {table} SET {field}= %s::jsonb->>{field} WHERE {field} IN %s".format(table=table, field=field)
+        ctype = column_type(cr, table, field)
+        if ctype == "varchar":
+            query = "UPDATE {table} t SET {column} = %(json)s::jsonb->>t.{column} WHERE t.{column} IN %(keys)s"
+        elif ctype == "jsonb":
+            # company dependent selection field
+            query = """
+                UPDATE {table}
+                   SET
+                    WITH upd AS (
+                         SELECT t.id,
+                                jsonb_object_agg(v.key, %s::jsonb->>{field}) AS value
+                           FROM {table} t
+                           JOIN LATERAL jsonb_each_text(t.{column}) v
+                             ON true
+                          WHERE jsonb_path_match(t.{column}, %(match)s)
```

Comment:
Can't we just do a simple check with keys?
```suggestion
                          WHERE v.value IN %(keys)s
```

Reply:
No, we need to keep other values. This (like the new version) is to filter out the records that should not be modified.